src/game/WorldHandlers/ChatHandler.cpp

```diff
@@ -256,7 +260,16 @@ void WorldSession::HandleMessagechatOpcode(WorldPacket& recv_data)
 #ifdef ENABLE_ELUNA
             sEluna->OnChat(GetPlayer(), type, lang, msg, player);
 #endif /* ENABLE_ELUNA */
-            GetPlayer()->Whisper(msg, lang, player->GetObjectGuid());
+#ifdef ENABLE_PLAYERBOTS
+            if (player->GetPlayerbotAI())
+            {
+                player->GetPlayerbotAI()->HandleCommand(type, msg, *GetPlayer());
+                GetPlayer()->m_speakTime = 0;
+                GetPlayer()->m_speakCount = 0;
+            }
+            else
+#endif
+                GetPlayer()->Whisper(msg, lang, player->GetObjectGuid());
```

Comment:
Maybe you could leave the default code alone and in the playerbot code use break; in the IF statement to exit before calling whisper. That way the code feels more clear to me, the indentation would be correct and there would be no dangling else.